src/components/MultiHopTrade/components/TradeConfirm/hooks/useActiveTradeAllowance.tsx

```diff
@@ -0,0 +1,121 @@
+import type { TradeQuoteStep } from '@shapeshiftoss/swapper'
+import { useCallback, useMemo } from 'react'
+import { useGetTradeQuotes } from 'components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes'
+import { AllowanceType } from 'hooks/queries/useApprovalFees'
+import { selectHopExecutionMetadata } from 'state/slices/tradeQuoteSlice/selectors'
+import { HopExecutionState } from 'state/slices/tradeQuoteSlice/types'
+import { useAppSelector } from 'state/store'
+
+import { useAllowanceApproval } from '../../MultiHopTradeConfirm/hooks/useAllowanceApproval'
+import { useAllowanceReset } from '../../MultiHopTradeConfirm/hooks/useAllowanceReset'
+import { useSignPermit2 } from '../../MultiHopTradeConfirm/hooks/useSignPermit2'
+import { useCurrentHopIndex } from './useCurrentHopIndex'
+
+type UseSignAllowanceApprovalProps = {
+  tradeQuoteStep: TradeQuoteStep
+  isExactAllowance: boolean | undefined
+  activeTradeId: string
+}
+
+export const useActiveTradeAllowance = ({
+  tradeQuoteStep,
+  isExactAllowance,
+  activeTradeId,
+}: UseSignAllowanceApprovalProps) => {
+  // TODO: confirm this is actually needed in the new flow
```

Comment:
We can remove that TODO, see https://github.com/shapeshift/web/issues/8269#issuecomment-2550744959 cc @NeOMakinG 

ZRX final quote will always be fetched at pre-permit2-signing time

Reply:
ty ser 🙏 https://github.com/shapeshift/web/commit/20c18f0b55d5c38c8aff1f9415eddde956816bf8